### PR TITLE
feat(dropdown): added dropdown option and cleaning up menu

### PR DESCRIPTION
**Pull request by willdyess**

### CODE REVIEW

File: projects/eamode/eang/themes/mode/menu.css [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
   & ea-menu {
+    --ea-menu-height: 2.5;
     user-select: none;
     cursor: pointer;
+    position: relative;
     & .node {
       display: flex;
-      line-height: 2.5em;
+      line-height: var(--ea-menu-height);
       align-items: center;
```
Comment (willdyess) [inline projects/eamode/eang/themes/mode/menu.css]: Fix this with fallback convention

File: projects/eamode/eang/src/components/menu.ts [hunk 2]
```diff
@@ -27,29 +28,39 @@ export interface MenuTreeItem {
...
-      <ng-container *ngIf="contentTemplate; else defTemplate"
-        [ngTemplateOutlet]="contentTemplate"
+    [attr.toggle]="node.toggleRight ? '' : null"
+    [class.dropdownNode]="node.dropdown">
+
+    <div
+    *ngIf="node.children?.length > 0" (click)="onToggle()"
+    [style.min-width]="!node.dropdown ? depth * 15 + 'px': null"
+    class="toggleArea"
+    [class.dropdownToggle]="node.dropdown"
```
Comment (sebastiankr) [inline projects/eamode/eang/src/components/menu.ts]: change to attr

File: projects/eamode/eang/src/components/menu.ts [hunk 3]
```diff
@@ -27,29 +28,39 @@ export interface MenuTreeItem {
...
+    >
+      <ng-container *ngIf="toggleAreaTemplate; else defaultTemplate"
+        [ngTemplateOutlet]="toggleAreaTemplate"
+        [ngTemplateOutletContext]="{node: node}">
+      </ng-container>
+      <ng-template #defaultTemplate>
+            <span icon chevron-down negative *ngIf="node.isOpen">
```
Comment (sebastiankr) [inline projects/eamode/eang/src/components/menu.ts]: #defaultToggleArea

File: projects/eamode/eang/src/components/menu.ts [hunk 4]
```diff
@@ -27,29 +28,39 @@ export interface MenuTreeItem {
...
+            <span icon chevron-left negative *ngIf="!node.isOpen && node.toggleRight">
+            </span>
+            <span icon chevron-right negative *ngIf="!node.isOpen && !node.toggleRight">
+            </span>
+      </ng-template>
+    </div>
+
+    <div
+    (click)="onActivate()"
+    class="name"
```
Comment (sebastiankr) [inline projects/eamode/eang/src/components/menu.ts]: make it attr and change it to name-area

File: projects/eamode/eang/src/components/menu.ts [hunk 5]
```diff
@@ -27,29 +28,39 @@ export interface MenuTreeItem {
...
+    [attr.toggle]="node.toggleRight ? '' : null"
+    [style.padding-left]="!node.children ? depth * 15 + 'px' : 0"
+    >
```
Comment (sebastiankr) [inline projects/eamode/eang/src/components/menu.ts]: reverse to positive and check logic for checking for children. should be checking for null AND length > 0?

File: projects/eamode/eang/src/components/menu.ts [hunk 6]
```diff
@@ -27,29 +28,39 @@ export interface MenuTreeItem {
...
+      <ng-container *ngIf="nameAreaTemplate; else defTemplate"
```
Comment (sebastiankr) [inline projects/eamode/eang/src/components/menu.ts]: rename deftemplate

File: projects/eamode/eang/src/components/menu.ts [hunk 7]
```diff
@@ -73,17 +84,21 @@ export interface MenuTreeItem {
     </aside>
 
 </div>
-<div *ngIf="node.children?.length > 0 && (node.isOpen || node.isHidden)" class="ea-tree-children" [class.horizontal]="!!node.horizontal">
+<div *ngIf="node.children?.length > 0 && (node.isOpen || node.isHidden)"
+class="ea-tree-children"
+[class.horizontal]="node.horizontal"
+[class.dropdown]="node.dropdown">
   <ea-menu
```
Comment (sebastiankr) [inline projects/eamode/eang/src/components/menu.ts]: classes?

File: projects/eamode/eang/src/components/tabs.ts [hunk 8]
```diff
@@ -14,7 +14,7 @@ import { TabComponent } from './tab'
   selector: 'ea-tabs',
   template: `
   <ea-menu [node]="menu" [activateEvents]="activated" [closeEvents]="closed"
-  [contentTemplate]="headerTemplate">
+  [nameAreaTemplate]="headerTemplate">
   </ea-menu>
```
Comment (sebastiankr) [inline projects/eamode/eang/src/components/tabs.ts]: all 3?

File: projects/eamode/eang/themes/mode/layout.css [hunk 9]
```diff
@@ -56,6 +56,7 @@
       margin: var(--ea-main-margin);
       display: var(--ea-body-display);
       min-height: 100%;
+      overflow: hidden;
     }
```
Comment (sebastiankr) [inline projects/eamode/eang/themes/mode/layout.css]: re-think overflow behavior

File: projects/eamode/eang/themes/mode/menu.css [hunk 10]
```diff
@@ -1,13 +1,20 @@
 @define-mixin menu {
   & ea-menu {
+    --ea-menu-height: 2.5;
     user-select: none;
     cursor: pointer;
+    position: relative;
     & .node {
```
Comment (sebastiankr) [inline projects/eamode/eang/themes/mode/menu.css]: could go into ea-menu.[dropdown]?

File: projects/eamode/eang/themes/mode/menu.css [hunk 11]
```diff
@@ -1,13 +1,20 @@
 @define-mixin menu {
   & ea-menu {
+    --ea-menu-height: 2.5;
```
Comment (sebastiankr) [inline projects/eamode/eang/themes/mode/menu.css]: should be em?

File: projects/eamode/eang/themes/mode/menu.css [hunk 12]
```diff
@@ -1,13 +1,20 @@
...
+    position: relative;
     & .node {
       display: flex;
-      line-height: 2.5em;
+      line-height: var(--ea-menu-height);
       align-items: center;
       position: relative;
 
+      & span[icon] {
+        height: 0.7rem;
```
Comment (sebastiankr) [inline projects/eamode/eang/themes/mode/menu.css]: should be em?

File: projects/eamode/eang/themes/mode/menu.css [hunk 13]
```diff
@@ -47,6 +50,26 @@
         );
         color: var(--ea-menu-color-active, var(--ea-text-color-light));
       }
+
+      &.dropdownNode{
+        &:hover {
+          background: transparent;
```
Comment (sebastiankr) [inline projects/eamode/eang/themes/mode/menu.css]: vars?

File: projects/eamode/eang/themes/mode/menu.css [hunk 14]
```diff
@@ -124,13 +147,31 @@
...
+      &.dropdown {
+        --ea-color: var(--ea-color-2);
+        position: absolute;
+        z-index: 1;
+        background: var(
+          --ea-background,
+          var(--ea-menu-background-dropdown, white)
+        );
+        min-width: 8rem;
+        box-shadow: var(--ea-box-shadow);
```
Comment (sebastiankr) [inline projects/eamode/eang/themes/mode/menu.css]: should be em

File: projects/eamode/eang/themes/mode/menu.css [hunk 15]
```diff
@@ -124,13 +147,31 @@
...
+        & .node {
+          & .name {
+            padding-left: 0 !important;
```
Comment (sebastiankr) [inline projects/eamode/eang/themes/mode/menu.css]: ouch!

**Pull request merged**